workflow/controller/operator.go

```diff
@@ -3397,23 +3356,5 @@ func (woc *wfOperationCtx) mergedTemplateDefaultsInto(originalTmpl *wfv1.Templat
 }
 
 func (woc *wfOperationCtx) substituteGlobalVariables() error {
-	execWfSpec := woc.execWf.Spec
-
-	// To Avoid the stale Global parameter value substitution to templates.
-	// Updated Global parameter values will be substituted in 'executetemplate' for templates.
-	execWfSpec.Templates = nil
-
-	wfSpec, err := json.Marshal(execWfSpec)
-	if err != nil {
-		return err
-	}
-	resolveSpec, err := template.Replace(string(wfSpec), woc.globalParams, true)
-	if err != nil {
-		return err
-	}
-	err = json.Unmarshal([]byte(resolveSpec), &woc.execWf.Spec)
-	if err != nil {
-		return err
-	}
-	return nil
+	return template.Replace(&woc.execWf.Spec, woc.globalParams, true)
```

Comment:
```suggestion
	execWfSpec := woc.execWf.Spec
	
	// To Avoid the stale Global parameter value substitution to templates.
	// Updated Global parameter values will be substituted in 'executetemplate' for templates.
	execWfSpec.Templates = nil
	
	return template.Replace(&woc.execWf.Spec, woc.globalParams, true)
```

@sarabala1979 like this?

Reply:
yes

Reply:
return template.Replace(&execWfSpec, woc.globalParams, true)